src/components/filter/MessagesFilterPanel.tsx

```diff
@@ -104,9 +106,16 @@ const MessagesFilterPanel = () => {
 		);
 	}, [filter, filterStore.filter, streams, isSoftFilterApplied]);
 
-	const isLoading = computed(
+	const stopLoading = React.useCallback(() => {
+		messagesDataStore.stopMessagesLoading();
+		workspaceStore.stopAttachedMessagesLoading();
+	}, []);
+
+	const isAnchorLoading = computed(() => workspaceStore.isLoadingAttachedMessages).get();
+
+	const isMessageListLoading = computed(
 		() =>
-			(messagesDataStore.messages.length === 0 && messagesDataStore.isLoading) ||
+			messagesDataStore.isLoading ||
```

Comment:
i think something is off with loading state, looks like it resets loading state before setting messages. And on initial loading there is some kind of bug, it shows that messages are loading, then it shows 'no messages' then spinner shows again